### PR TITLE
Finished off a large chunk of the opcodes

### LINKED ISSUE #7: Implement Opcode 1
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #9: Implement opcode 3
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #10: Implement opcode 4
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #11: Implement opcode 5
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #12: Implement opcode 6
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #13: Implement opcode 7
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #16: Implement opcode A
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #18: Implement opcode C
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### LINKED ISSUE #19: Implement opcode D
See https://en.wikipedia.org/wiki/CHIP-8

Implement
Confirm it works

### PR DESCRIPTION
**Pull request by Cryowatt**
Most of the opcodes are implemented now. Op 8 is still missing most of it's sub-operations.

Currently the breakout board is correctly rendered:
image

The next instruction (#$FF0A) blocks execution until the next keypress, so next session we'll have to implement some keyboard input. Most people seem to map the hex keypad as follows:

```
[1][2][3][4] >> [C][D][E][F]
[Q][W][E][R] >> [8][9][A][B]
[A][S][D][F] >> [4][5][6][7]
[Z][X][C][V] >> [0][1][2][3]
```

Closes #7
Closes #9
Closes #10
Closes #11
Closes #12
Closes #13
Closes #16
Closes #18
Closes #19

### CODE REVIEW

**Review by davemeyer — APPROVED** [review]
🚢 🇮🇹 f

File: Hahnium.Chip8/Chip8Cpu.cs [hunk 1]
```diff
@@ -39,45 +47,169 @@ public unsafe void Cycle()
...
-            // Sets I to the address NNN
-            this.address = operands;
+            if (operands == 0x00e0)
+            {
+                //Clears the screen.
+                Console.Clear();
+            }
+            else if (operands == 0x00ee)
+            {
+                // TODO: return
```
Comment (mpaauw) [inline Hahnium.Chip8/Chip8Cpu.cs:90]: Do we need separate issues open for remaining implementations?

File: Hahnium.Chip8/Chip8Ppu.cs [hunk 2]
```diff
@@ -49,14 +50,14 @@ public Chip8Ppu(Chip8Cpu cpu, Memory<byte> ram)
...
             // Test animation
-            for (int y = 0; y < DisplayHeight; y++)
-            {
-                for (int x = 0; x < DisplayWidth; x++)
-                {
-                    int offset = (y * DisplayWidth) + x;
-                    frameBuffer[offset] = (byte)(((offset + cycleCount + y) % 8) == 1 ? 1 : 0);
-                }
-            }
+            //for (int y = 0; y < DisplayHeight; y++)
```
Comment (mpaauw) [inline Hahnium.Chip8/Chip8Ppu.cs]: Remove unused code block.

File: Hahnium.Chip8/Chip8Ppu.cs [hunk 3]
```diff
@@ -81,25 +82,54 @@ internal void Cycle()
...
+            var sprite = (byte*)spriteHandle.Pointer;
+
+            for (int y = yPosition; y < yPosition + spriteMemory.Length; y++)
+            {
+                byte spriteLine = *sprite;
+
+                for (int x = xPosition; x < xPosition + 8; x++)
+                {
+                    byte pixel = frameBuffer[(y * DisplayWidth) + x];
+                    var spritePixel = (spriteLine & 0b1000_0000) >> 7;
```
Comment (mpaauw) [inline Hahnium.Chip8/Chip8Ppu.cs]: Maybe remove binary literals to constants to make it more clear what's going on?

**Review by mpaauw — CHANGES_REQUESTED** [review]

File: Hahnium.Chip8/Chip8Cpu.cs (hunk 1 above)
Comment (Cryowatt) [reply Hahnium.Chip8/Chip8Cpu.cs:90]: Maybe just bullet points in the existing Op0 issue. Most of the remaining op work is a line or two of code per unique instruction

File: Hahnium.Chip8/Chip8Ppu.cs (hunk 2 above)
Comment (Cryowatt) [reply Hahnium.Chip8/Chip8Ppu.cs]: Mite move it to it's own function. It's handy enough to keep around for debug/testing.

File: Hahnium.Chip8/Chip8Ppu.cs (hunk 3 above)
Comment (Cryowatt) [reply Hahnium.Chip8/Chip8Ppu.cs]: Personally I think it's clearer like this. I want to grab the most significant bit and shift it down to a 0/1 value. Would it be clearer for everyone to rename it MSB (most significant bit)?

Comment (mpaauw) [reply Hahnium.Chip8/Chip8Ppu.cs]: Yeah I think so.

**Pull request merged**